src/libsyntax/parse/lexer/mod.rs

```diff
@@ -159,11 +165,30 @@ impl<'a> StringReader<'a> {
         self.ch.is_none()
     }
 
-    fn fail_unterminated_raw_string(&self, pos: BytePos, hash_count: u16) {
-        let mut err = self.struct_span_fatal(pos, pos, "unterminated raw string");
-        err.span_label(self.mk_sp(pos, pos), "unterminated raw string");
+    fn fail_unterminated_raw_string(&self, start: Span, hash_count: u16, spans: Vec<Span>) -> ! {
+        const SPAN_THRESHOLD: usize = 3;
+        const MSG_STR: &str = "Raw string could be meant to end here";
```

Comment:
Unfortunately I don't really have a better suggestion yet, but this message feels like something a native English speaker wouldn't say.  (Which I'm most certainly not, so I might as well be wrong about this.)

Reply:
"you might have meant to end the raw string here"